dot/rpc/modules/sync_state.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package modules
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+)
+
+type SyncStateModule struct {
+	SyncStateAPI SyncStateAPI
+}
+
+type BoolRequest struct {
+	Raw bool
+}
+
+func NewSyncStateModule(s SyncStateAPI) *SyncStateModule {
+	return &SyncStateModule{SyncStateAPI: s}
+}
+
+func (ss *SyncStateModule) GenSyncSpec(_ *http.Request, req *BoolRequest, res *genesis.Genesis) error {
```

Comment:
IMO the `SyncStateModule` does not need to receive an interface as a parameter, I think could be better to receive the `GenesisFilePath` as a string parameter and move the implementation of `GenSyncSpec`as an exported function at `lib/genesis/genesis.go`. And in the RPC module implementation just call the `genesis.GenSyncSpec(ss.GenesisFilePath)`